ports/servo/resources.rs

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use servo::embedder_traits::resources::{self, Resource};
+
+struct ResourceReader;
+
+fn filename(file: Resource) -> &'static str {
+    match file {
+        Resource::Preferences => "prefs.json",
+        Resource::BluetoothBlocklist => "gatt_blocklist.txt",
+        Resource::DomainList => "public_domains.txt",
+        Resource::HstsPreloadList => "hsts_preload.json",
+        Resource::SSLCertificates => "certs",
+        Resource::BadCertHTML => "badcert.html",
+        Resource::NetErrorHTML => "neterror.html",
+        Resource::UserAgentCSS => "user-agent.css",
+        Resource::ServoCSS => "servo.css",
+        Resource::PresentationalHintsCSS => "presentational-hints.css",
+        Resource::QuirksModeCSS => "quirks-mode.css",
+        Resource::RippyPNG => "rippy.png",
+    }
+}
+
+pub fn init() {
+    resources::set(Box::new(ResourceReader));
+}
+
+#[cfg(not(target_os = "android"))]
+mod not_android {
+    use servo::embedder_traits::resources::{self, Resource};
+    use std::env;
+    use std::fs::File;
+    use std::io::{self, Read};
+    use std::path::PathBuf;
+    use std::sync::Mutex;
+
+    lazy_static! {
+        static ref CMD_RESOURCE_DIR: Mutex<Option<String>> = Mutex::new(None);
+    }
+
+    impl resources::ResourceReaderMethods for super::ResourceReader {
+        fn read(&self, file: Resource) -> Vec<u8> {
+            let file = super::filename(file);
+            let mut path = resources_dir_path().expect("Can't find resources directory");
+            path.push(file);
+            let mut buffer = vec![];
+            File::open(path).expect(&format!("Can't find file: {}", file))
+                .read_to_end(&mut buffer).expect("Can't read file");
+            buffer
+        }
+        fn sandbox_access_files_rec(&self) -> Vec<PathBuf> {
+            vec![resources_dir_path().expect("Can't find resources directory")]
```

Comment:
If this just returns paths to top-level directories (rather than recursively iterating over the files inside those directories), maybe it should be renamed to `sandox_access_files_dirs`.